addons/account/tests/test_account_payment.py

```diff
@@ -903,3 +903,105 @@ def test_internal_transfer_right_accounts(self):
+            'payment_type': 'inbound',
+            'partner_type': 'customer',
+            'payment_reference': False,
+            'is_reconciled': False,
+            'currency_id': self.company_data['currency'].id,
+            'partner_id': self.partner_a.id,
+            'journal_id': journal_A.id
+        }
+        expected_move_values = {
+            'currency_id': self.company_data['currency'].id,
+            'partner_id': self.partner_a.id,
+            'journal_id': journal_A.id,
+        }
+        expected_liquidity_line = {
+            'debit': 50.0,
+            'credit': 0.0,
+            'amount_currency': 50.0,
+            'currency_id': self.company_data['currency'].id,
+            'account_id': outstanding_payment_A.id,
+        }
+        expected_counterpart_line = {
+            'debit': 0.0,
+            'credit': 50.0,
+            'amount_currency': -50.0,
+            'currency_id': self.company_data['currency'].id,
+            'account_id': self.company_data['default_account_receivable'].id,
+        }
+        self.assertRecordValues(payment, [expected_payment_values])
+        self.assertRecordValues(payment.move_id, [expected_move_values])
+        self.assertRecordValues(payment.line_ids.sorted('balance'), [
+            expected_counterpart_line,
+            expected_liquidity_line,
+        ])
+
+        # Change the journal on the form
+        pay_form.journal_id = journal_B
+        # Save the form (to write move and move line)
+        payment = pay_form.save()
+
+        # Check the payment
+        expected_payment_values = {
+            'amount': 50.0,
+            'payment_type': 'inbound',
+            'partner_type': 'customer',
+            'payment_reference': False,
+            'is_reconciled': False,
+            'currency_id': self.company_data['currency'].id,
+            'partner_id': self.partner_a.id,
+            'journal_id': journal_B.id
+        }
+        expected_move_values = {
+            'currency_id': self.company_data['currency'].id,
+            'partner_id': self.partner_a.id,
+            'journal_id': journal_B.id,
+        }
+        expected_liquidity_line = {
+            'debit': 50.0,
+            'credit': 0.0,
+            'amount_currency': 50.0,
+            'currency_id': self.company_data['currency'].id,
+            'account_id': outstanding_payment_B.id,
+        }
+        expected_counterpart_line = {
+            'debit': 0.0,
+            'credit': 50.0,
+            'amount_currency': -50.0,
+            'currency_id': self.company_data['currency'].id,
+            'account_id': self.company_data['default_account_receivable'].id,
+        }
+        self.assertRecordValues(payment, [expected_payment_values])
+        self.assertRecordValues(payment.move_id, [expected_move_values])
+        self.assertRecordValues(payment.line_ids.sorted('balance'), [
+            expected_counterpart_line,
+            expected_liquidity_line,
```

Comment:
why do you need to create temporary variable for that?
```py
        self.assertRecordValues(payment, [{
            'amount': 50.0,
            'payment_type': 'inbound',
            'partner_type': 'customer',
            'payment_reference': False,
            'is_reconciled': False,
            'currency_id': self.company_data['currency'].id,
            'partner_id': self.partner_a.id,
            'journal_id': journal_B.id
        }])
        self.assertRecordValues(payment.move_id, [{
            'currency_id': self.company_data['currency'].id,
            'partner_id': self.partner_a.id,
            'journal_id': journal_B.id,
        }])
        self.assertRecordValues(payment.line_ids.sorted('balance'), [
            {
                'debit': 0.0,
                'credit': 50.0,
                'amount_currency': -50.0,
                'currency_id': self.company_data['currency'].id,
                'account_id': self.company_data['default_account_receivable'].id,
            },
            {
                'debit': 50.0,
                'credit': 0.0,
                'amount_currency': 50.0,
                'currency_id': self.company_data['currency'].id,
                'account_id': outstanding_payment_B.id,
            },
        ])
```

Reply:
It was for readability but not essential. I will correct this.